internal/worker/worker.go

```diff
@@ -240,6 +250,39 @@ func (w *Worker) executeTask(ctx context.Context, t *tork.Task) error {
 	return nil
 }
 
+func (w *Worker) createMount(ctx context.Context, spec string) (mount string, delete func() error, err error) {
```

Comment:
I'm not sure if implementing your own logic here may be a good idea. The Docker golang package has a lib for handling mounts already. For example look at the one from testcontainers-go

https://github.com/testcontainers/testcontainers-go/blob/main/docker_mounts.go

Reply:
There's so many edge cases, etc that could happen too

Reply:
For sure, but the custom logic is really just to guard against the "illegal" mounts which the SDK isn't going to do for me. The actual mounting is delegated to the Docker SDK: https://github.com/runabol/tork/blob/d69c8364e8a43cbc83dce8deda24be5c765bae28/runtime/docker.go#L163-L185

Reply:
I could rename the method name though. it's probably misleading.

Reply:
Yes, haha

Reply:
That's fair. I'll do that.

Reply:
Besides that it looks overall good to me. May be worth in the future adding an option in the `Task` yaml for creating `tmpfs`. Example:

```yaml
name: my task
image: some:image
volumes:
  - /some/local/dir:/target/mount
tmpfs:
  - /cache
```

Reply:
Yes, the trouble with tmpfs is that it's only supported on Linux hosts and also it clears when the container stops according to the Docker docs: https://docs.docker.com/storage/tmpfs/ which won't be helpful for caches between tasks. 

How do you feel about this:

```
 name: my task
 image: some:image
 run: ...
 volumes:
   - /myvolume
 mounts:
   - /local/dir:/mount/point
 cache: 
   - /go/pkg
```

To make it extra clear the intent between the different types of mounts?

The first two are pretty self explanatory. 

The idea with `cache` is that it's an ordinary volume but it will survive the execution of a container, but is scoped for the lifetime of the job (because I don't think it's a good idea to share state across jobs. That's a recipe for a mess).

Reply:
I like that approach, it's more straight-forward and UX friendly. I didn't know `tmpfs` is only supported on Linux (it's the only thing I use for code).

At least for me I heavily use `tmpfs` to avoid writing temp data to disk. For example, let's say I have a service that grabs data from a folder/remote location and creates a Tar. Normally I would write that Tar into a tmpfs first before posting it upstream.  

But yeah, if they are not supported in anything other than Linux, it doesn't make sense to add them if the intent is to be cross-platform.

Reply:
I do agree with you though on the utility of tmpfs. It's really fast too. Maybe I add it in a separate PR with a note on the documentation that it's only supported on Linux hosts. I think that's a fair caveat.

Reply:
Yes, that should work. That's what Docker did 😆